sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -604,6 +604,8 @@ case class Least(children: Seq[Expression]) extends Expression {
     val evalChildren = children.map(_.genCode(ctx))
     val first = evalChildren(0)
```

Comment:
what about avoiding splitting `first` and `rest` and treat them all the same? Since the declaration of `ev.isNull` and `ev.value` is moved, they can all be treated the same way. In this way the code would be cleaner, wouldn't it?

Reply:
Do you want to assign `evalChildren(0).value` into `ev.value` and then compare all of `evalChildren`? It can avoid `rest` for cleaner code. However, it increases one comparison.  
Or, do you have another idea?

Reply:
I am thinking about doing what I did in #19720: here we might follow the same approach. Also for consistency across the code. Indeed, there would be an `if` more, but I think that the overhead would be negligible.
Or if we state that the right approach is this one to avoid one more `if`, I am best to change my approach in the PR I mentioned.

Reply:
I know what you did. This PR cannot follow the approach straightforward due to initial value. We can easily know `min`/`max` value for primitive type for an initial value. How can we know the initial value for complicated type?
IMHO, the first element would be good for an initial value.

What do you think?

Reply:
I don't see any difference on the initial value. IMHO the initial value in my PR as well as here should be "null",  which means `ev. isNull` to `true` and the initial value for `ev.value` is irrelevant and it could be set to what is returned by the `defaultValue` function of the `CodegenContext`. Do you have any concern with this?

Reply:
[`defaultValue` function](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala#L596) returns `-1` or `null`.

When we compare any non-primitive value with `null`, `NullPointerException` may occur.  
Wehn we compare primitive value with initial value `-1`, the result may cause an incorrect result. For example, in `Greatest`, if all of the given values are `-2`, the result will be `-1`. It is incorrect.

What do you think?

Reply:
the `value` is used only when `isNull` is `false`. And in that context a proper value has been set for `value`, thus the problem you are describing can't happen IMHO.

Reply:
I am sorry that I cannot understand your opinion.
Could you please explain how your opinion works in the case that all of childrens with `ev.value = -2` and `ev.isNull = false` for `Greatest` by using `ctx.defaultValue`?

Reply:
before using `ev.value`, `ev.isNull` is checked (https://github.com/kiszk/spark/blob/bc3f0a43956a1e46928f0269d1d8d0aad9ee02b1/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala#L612). So in case of all children with `-2`, the first one will see that `ev.isNull` is `true` and then it will set it to `false` and assign `-2` to `ev.value`. Thus the initial value assigned to `ev.value` is irrelevant, since `ev.isNull` is `true` and as far as `ev.isNull` is `true`, what `ev.value` contains doesn't matter.

Reply:
Thank you for your explanation. Got it.  
Your idea is to set `true` into `ev.isNull` rather than using `ctx.defaultValue` as an initial value for `ev.value`.

Reply:
yes exactly, it doesn't matter how `ev.value` is initialized.